src/nrf_802154_types.h

```diff
@@ -138,6 +138,15 @@ typedef uint8_t nrf_802154_ack_data_t;
 #define NRF_802154_ACK_DATA_PENDING_BIT 0x00
 #define NRF_802154_ACK_DATA_IE          0x01
 
+/**
+ * @brief Methods of source address matching.
+ */
+typedef uint8_t nrf_802154_src_match_t;
```

Comment:
I think it would be really, really nice if the name of this type informed explicitly that it's about the source *address*, which is very important and not necessarily obvious now. `nrf_802154_src_addr_match_t` ?